python/ray/serve/master.py

```diff
@@ -424,6 +426,25 @@ def _scale_replicas(self, backend_tag, num_replicas):
         current_num_replicas = len(self.replicas[backend_tag])
         delta_num_replicas = num_replicas - current_num_replicas
 
+        _, _, replicas_config = self.backends[backend_tag]
+        resource_per_replica = replicas_config.resource_dict
+
+        if delta_num_replicas > 0:
+            scheduler = MockScheduler()
+            can_schedule = [
+                scheduler.try_schedule(resource_per_replica)
+                for _ in range(delta_num_replicas)
+            ]
+            if not all(can_schedule):
+                num_possible = sum(can_schedule)
+                raise RayServeException(
```

Comment:
Try improve the UX I think we should throw an exception here and then catch it in the API client to avoid the huge `RayTaskError` callback.

Reply:
Turns out to be kind of ugly to do since we need to parse the string traceback and figure out how to re-raise it. Adding to the backlog and not doing it in this PR.